ocaml/xapi/xapi_globs.ml

```diff
@@ -466,6 +466,11 @@ let redo_log_alert_key = "metadata_lun_alerts"
 let serialize_pool_enable_disable_extauth = Mutex.create()
 (* CP-695: controls our asynchronous persistent initialization of the external authentication service during Xapi.server_init *)
 
+(* Auth types *)
+let auth_type_NONE = ""
+let auth_type_AD_Likewise = "AD"
+let auth_type_PAM = "PAM"
```

Comment:
I would prefer a variant type with a `auth_type_of_string` function to just 3 string values. It's a shame it's not encoded like that in the database.

Could the database (de)serialization be changed to use this?

Reply:
yes, this is what I am thinking at first,
then we need to (de)serialization database values. 
It is a kind of code refine, and I just focus on issues in the ticket itself.